src/lualib/ArrayReduce.ts

```diff
@@ -1,27 +1,35 @@
+/** @vararg */
+interface Vararg<T> extends Array<T> {}
+/** @forRange */
+declare function forRange(start: number, limit: number, step?: number): number[];
+
 // https://www.ecma-international.org/ecma-262/9.0/index.html#sec-array.prototype.reduce
 function __TS__ArrayReduce<T>(
     this: void,
     arr: T[],
     callbackFn: (accumulator: T, currentValue: T, index: number, array: T[]) => T,
-    initial?: T
+    ...initial: Vararg<T>
 ): T {
     const len = arr.length;
 
-    if (len === 0 && initial === undefined) {
+    const initialValuePresent = select("#", ...initial) !== 0;
+    if (len === 0 && !initialValuePresent) {
```

Comment:
Thoughts on moving that check to a condition below and removing `initialValuePresent` variable?